tests/data/sentinel2/data.py

```diff
@@ -154,18 +219,33 @@ def create_file(path: str, dtype: str, num_channels: int) -> None:
     profile['driver'] = 'JP2OpenJPEG'
     profile['dtype'] = dtype
     profile['count'] = num_channels
-    profile['crs'] = CRS.from_epsg(32616)
+    crs = CRS.from_epsg(32616)
+    profile['crs'] = crs
     profile['transform'] = Affine(res, 0.0, 399960.0, 0.0, -res, 4500000.0)
-    profile['height'] = round(SIZE * 10 / res)
-    profile['width'] = round(SIZE * 10 / res)
+    raster_height = round(SIZE * 10 / res)
+    raster_width = round(SIZE * 10 / res)
+    profile['height'] = raster_height
+    profile['width'] = raster_width
+    # NB! .SAFE format does not include nodata value in the raster profile...
 
     if 'float' in profile['dtype']:
-        Z = np.random.randn(SIZE, SIZE).astype(profile['dtype'])
+        Z = np.random.randn(raster_height, raster_width).astype(profile['dtype'])
     else:
         Z = np.random.randint(
-            np.iinfo(profile['dtype']).max, size=(SIZE, SIZE), dtype=profile['dtype']
```

Comment:
I think this has been wrong. The contents of the raster should match the width and height of the raster definition, right?